port/unix/omrsysinfo.c

```diff
@@ -3948,6 +3987,67 @@ getHandleOfCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subs
 	return rc;
 }
 
+/**
+ * Reads the Cgroup metric (if has multiple searched for metric key and reads it) and returns the value to the param `value`
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[in] subsystemFlag flag of type OMR_CGROUP_SUBSYSTEMS_* representing the cgroup subsystem
+ * @param[in] fileName name of the file under cgroup subsystem
+ * @param[in] metricKeyInFile name of the cgroup metric which is in a group of metrics in a cgroup subsystem file
+ * @param[in/out] value pointer to char * which stores the value for the specified cgroup subsystem metric
+ *
+ * @return 0 on success, negative error code on any error
+ */
+static int32_t
+readCgroupMetricFromFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFlag, const char *fileName, const char *metricKeyInFile, char *value)
+{
+	int32_t rc = 0;
+	FILE *file = NULL;
+	rc = getHandleOfCgroupSubsystemFile(portLibrary, subsystemFlag, fileName, &file);
+	if (0 != rc) {
+		goto _end;
+	}
+	if (NULL == metricKeyInFile) { /* If `metricKeyInFile` is NULL then it states the file has only one Cgroup metric to be read */
+		if (NULL != fgets(value, MAX_LINE_LENGTH, file)) {
```

Comment:
As we pass the value array of `OMRCgroupMetricElement` i thought i could write it that way. I will send the check value too in the function params so that it would be generic to check in any context other than `MAX_LINE_LENGTH`